dwave/cloud/config/models.py

```diff
@@ -231,18 +231,28 @@ def validate_config_v1(raw_config: abc.Mapping) -> ClientConfig:
         try:
             solver_def = orjson.loads(solver)
         except Exception:
-            # unparseable (or non-dict) json, assume string name for solver
-            # we'll deprecate this eventually, but for now just convert it to
-            # features dict (equality constraint on full solver name)
-            logger.debug("Invalid solver JSON, assuming string name: %r", solver)
-            solver_def = dict(name__eq=solver)
+            # unparseable (or non-dict) json, assume solver identity as string
+            logger.info("Invalid solver JSON, parsing as string identity: %r", solver)
+            try:
+                identity = _solver_id_as_identity(solver)
+            except Exception:
+                logger.debug("Unknown solver identity string format, falling back to string")
+                solver_def = dict(id__eq=solver)
+            else:
+                # use identity equality constraint only when full identity is indeed
+                # specified; this is to prevent mismatches when user specifies just a name
+                # (which is a partial identity for a structured solver)
+                if identity.get('version'):
+                    solver_def = dict(identity__eq=identity)
+                else:
+                    solver_def = dict(name__eq=identity['name'])
```

Comment:
Not too familiar with this, but would setting `name__eq` be unnecessary if `identity__eq` is there?